paymentsheet/src/main/java/com/stripe/android/link/ui/updatecard/UpdateCardScreen.kt

```diff
@@ -2,28 +2,50 @@ package com.stripe.android.link.ui.updatecard
 
 import androidx.compose.foundation.layout.padding
 import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import com.stripe.android.DefaultCardBrandFilter
+import com.stripe.android.core.model.CountryCode
+import com.stripe.android.link.theme.DefaultLinkTheme
+import com.stripe.android.link.theme.StripeThemeForLink
 import com.stripe.android.link.ui.PrimaryButton
 import com.stripe.android.link.ui.PrimaryButtonState
 import com.stripe.android.link.ui.ScrollableTopLevelColumn
 import com.stripe.android.link.ui.SecondaryButton
+import com.stripe.android.model.CardBrand
+import com.stripe.android.model.ConsumerPaymentDetails
+import com.stripe.android.model.CvcCheck
+import com.stripe.android.paymentsheet.PaymentSheet.BillingDetailsCollectionConfiguration.AddressCollectionMode
 import com.stripe.android.paymentsheet.R
+import com.stripe.android.paymentsheet.ui.CardDetailsEditUI
+import com.stripe.android.paymentsheet.ui.DefaultEditCardDetailsInteractor
+import com.stripe.android.paymentsheet.ui.EditCardDetailsInteractor
+import com.stripe.android.paymentsheet.ui.EditCardPayload
 import com.stripe.android.R as StripeR
+import com.stripe.android.uicore.utils.collectAsState
 
 @Composable
 internal fun UpdateCardScreen(viewModel: UpdateCardScreenViewModel) {
+    val state by viewModel.state.collectAsState()
     UpdateCardScreenBody(
+        interactor = viewModel.interactor,
+        isDefault = state?.paymentDetails?.isDefault == true,
         onUpdateClicked = viewModel::onUpdateClicked,
         onCancelClicked = viewModel::onCancelClicked,
     )
 }
 
 @Composable
 internal fun UpdateCardScreenBody(
+    interactor: EditCardDetailsInteractor,
+    isDefault: Boolean,
```

Comment:
We should also pass in the enabled state of the primary button, and any processing state. This can happen in a follow-up.

Reply:
yep, coming on a followup!